ledger-core/network/transport/fake.go

```diff
@@ -140,3 +145,25 @@ func (f *fakeStreamTransport) Dial(ctx context.Context, address string) (io.Read
 func (f *fakeStreamTransport) Address() string {
 	return f.address
 }
+
+// WaitFakeListeners wait for streamTransports will be started
+// this is used to prevent race in tests
+func WaitFakeListeners(count int32, timeout time.Duration) error {
+	c := make(chan error)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	go func(ctx context.Context) {
+		for ;atomic.LoadInt32(&listenersCount) != count && ctx.Err() == nil; {
+			<-time.After(time.Microsecond*100)
```

Comment:
not critical for tests, but better to always avoid using time.After without select from multiple channels
because time.Sleep is cheaper - it doesn't trash heap & scheduler

Reply:
fixed